experimental/sql/src/fuse_proj_into_scan.py

```diff
@@ -25,29 +25,45 @@ def match_and_rewrite(self, op: RelImpl.Project, rewriter: PatternRewriter):
     if not isinstance(op.input.op, RelImpl.FullTableScanOp):
       return
 
-    # This rewrite expects that the projection only maps onto a subset of the
-    # columns of the input. Therefore, we first check that only IndexByName is
-    # present and then that there are equally as many cols that are yielded as
-    # there are in the result.
-    if not all([
-        isinstance(o.op, RelImpl.IndexByName) for o in op.projection.ops[-1].ops
-    ]):
-      return
-
     assert isinstance(op.projection.ops[-1], RelImpl.YieldTuple)
     yielded_ssa_values = op.projection.ops[-1].ops
 
-    col_names = [o.op.col_name.data for o in yielded_ssa_values]
-    res_names = [s.elt_name.data for s in op.input.typ.schema.data]
-    if len(col_names) == len(res_names):
+    # Get the list of used columns. The dict magic removes duplicates.
```

Comment:
Isn't this the typical idiom: `list(set([1,1,1,1,1]))`?

Reply:
Ah, and then you can avoid the construction of the inner list by making it a generator expression: `list(set((1 for i in range(4))))`

Reply:
I think the main difference is that the dict version keeps the order, that's why its my default go to. Order doesn't matter here though, so I am fine with changing to set, if you feel its necessary.

Reply:
Ah, good point about the order. No, pick whatever you think fits best. (I think the generator expression in favor of the list comprehension doesn't have any disadvantages, though.)